Purchases/Networking/Backend.swift

```diff
@@ -495,14 +495,16 @@ private extension Backend {
         }
 
         do {
-            let customerInfo = try CustomerInfo(data: response)
+            let customerInfo = try parseCustomerInfo(fromMaybeResponse: maybeResponse)
             let created = statusCode == HTTPStatusCodes.createdSuccess.rawValue
             Logger.user(Strings.identity.login_success)
             completion(customerInfo, created, nil)
-        } catch {
+        } catch let customerInfoError {
             Logger.error(Strings.backendError.customer_info_instantiation_error(maybeResponse: response))
+            let extraContext = "statusCode: \(statusCode), json:\(maybeResponse.debugDescription)"
             let subErrorCode = UnexpectedBackendResponseSubErrorCode.loginResponseDecoding
-            let responseError = ErrorUtils.unexpectedBackendResponse(withSubError: subErrorCode)
+            let responseError = ErrorUtils.unexpectedBackendResponse(withSubError: customerInfoError,
+                                                                     extraContext: extraContext)
```

Comment:
I can't "suggest changes" due to combining additions and deletions, but this is a great catch, thank you for digging into it.

Note:
`parseCustomerInfo` will create an `Error` that contains the entire response json, if it is available. It will attach it as `extraContext` to the `customerInfoError`  so adding it at this level isn't needed 😄

Here's a small change you can make to the PR, and then it will be 💯

Replace the whole `catch` block with the below:

```
            Logger.error(Strings.backendError.customer_info_instantiation_error(maybeResponse: response))
            let extraContext = "statusCode: \(statusCode)"
            let subErrorCode = UnexpectedBackendResponseSubErrorCode
                .loginResponseDecoding
                .addingUnderlyingError(customerInfoError)
            let responseError = ErrorUtils.unexpectedBackendResponse(withSubError: subErrorCode,
                                                                     extraContext: extraContext)
            completion(nil, false, responseError)
```

I think this captures and packages up the Errors nicely - Here's the hierarchy:
(`.unexpectedBackendResponse` + `statusCode`) + underlying `.loginResponseDecoding` + underlying  `customerInfoError` + extraContext (including json)

Reply:
Perfect! Makes sense.